src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp

```diff
@@ -362,41 +282,66 @@ enum Kalec
 
 struct boss_kalec : public ScriptedAI
 {
-    boss_kalec(Creature* creature) : ScriptedAI(creature) { }
-
-    EventMap events;
-    EventMap events2;
+    boss_kalec(Creature* creature) : ScriptedAI(creature)
+    {
+        SetInvincibility(true);
+    }
 
     void Reset() override
     {
-        events.Reset();
-        events2.Reset();
         if (me->GetPositionY() < 750.0f)
         {
             me->SetSpeed(MOVE_RUN, 2.4f);
             me->SetDisplayId(MODEL_KALECGOS_DRAGON);
             me->SetDisableGravity(true);
             me->GetMotionMaster()->MovePoint(0, 1483.30f, 657.99f, 28.0f, false, true);
-            events2.ScheduleEvent(EVENT_KALEC_SCENE_1, 9000);
-            events2.ScheduleEvent(EVENT_KALEC_SCENE_2, 16000);
-            events2.ScheduleEvent(EVENT_KALEC_SCENE_3, 22000);
+
+            me->m_Events.AddEventAtOffset([&] {
+                Talk(SAY_GOOD_MADRIGOSA);
+                me->GetMotionMaster()->MovePoint(0, 1509.0f, 560.0f, 30.0f, false, true);
+            }, 9s);
+
+            me->m_Events.AddEventAtOffset([&] {
+                DoCastAOE(SPELL_OPEN_BRUTALLUS_BACK_DOOR, true);
+                me->GetInstanceScript()->SetBossState(DATA_FELMYST_DOORS, NOT_STARTED);
+                me->GetInstanceScript()->SetBossState(DATA_FELMYST_DOORS, DONE);
+            }, 16s);
+
+            me->m_Events.AddEventAtOffset([&] {
+                me->GetMotionMaster()->MovePoint(0, 1400.0f, 630.0f, 90.0f, false, true);
+                me->DespawnOrUnsummon(6000);
+            }, 22s);
         }
         else
-            me->CastSpell(me, SPELL_SPECTRAL_INVISIBILITY, true);
-    }
-
-    void DamageTaken(Unit* who, uint32& damage, DamageEffectType, SpellSchoolMask) override
-    {
-        if (!who || who->GetEntry() != NPC_SATHROVARR)
-            damage = 0;
+            DoCastSelf(SPELL_SPECTRAL_INVISIBILITY, true);
     }
 
     void JustEngagedWith(Unit*) override
     {
-        events.ScheduleEvent(EVENT_CHECK_HEALTH, 1000);
-        events.ScheduleEvent(EVENT_CHECK_HEALTH2, 1000);
-        events.ScheduleEvent(EVENT_SPELL_REVITALIZE, 5000);
-        events.ScheduleEvent(EVENT_SPELL_HEROIC_STRIKE, 3000);
+        ScheduleTimedEvent(5s, [&] {
+            DoCastSelf(SPELL_REVITALIZE);
+        }, 10s);
+
+        ScheduleTimedEvent(3s, [&] {
+            DoCastVictim(SPELL_HEROIC_STRIKE);
+        }, 5s);
+
+        scheduler.Schedule(1s, [this](TaskContext context)
```

Comment:
Why are you not using ScheduleHealthCkeckEvent() here? Is this only available in BossAI?

Reply:
Yes